noir-projects/noir-protocol-circuits/crates/private-kernel-lib/src/private_kernel_circuit_public_inputs_composer.nr

```diff
@@ -15,6 +15,7 @@ use dep::types::{
 
 struct DataSource {
     private_call_public_inputs: PrivateCircuitPublicInputs,
+    contract_address: AztecAddress,
```

Comment:
Why was this new field needed? Isn't it the same as `storage_contract_address` right below?

Reply:
I was going off the comment over in `private_call_stack_item.nr`:
>    // This is the _actual_ contract address relating to where this function's code resides in the
    // contract tree. Regardless of whether this is a call or delegatecall, this
    // `contract_address` _does not change_. Amongst other things, it's used as a lookup for
    // getting the correct code from the tree. There is a separate `storage_contract_address`
    // within a CallStackItem which varies depending on whether this is a call or delegatecall.
    contract_address: AztecAddress,

Reply:
But then you want to use the `storage_contract_address` for fee payment, because you want the balance to be taken from the state of that address, regardless of the code being run.

Reply:
So in the case of delegate call, we're saying `contract_address` will be the address of the delgatee? But if `is_fee_payer` is set during that delegate call, we want it to be the delegator, which will be `storage_contract_address`?

Reply:
Okay updated the code and spec.